src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1900 @@ interface MPSMatrixMultiplication {
+		IntPtr Constructor (IMTLDevice device, MPSNnImageNode resultImage);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("sourceImageHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] SourceImageHandles { get; }
+
+		[NullAllowed, Export ("sourceStateHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] SourceStateHandles { get; }
+
+		[NullAllowed, Export ("intermediateImageHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] IntermediateImageHandles { get; }
+
+		[NullAllowed, Export ("resultStateHandles", ArgumentSemantic.Copy)]
+		IMPSHandle[] ResultStateHandles { get; }
+
+		[NullAllowed, Export ("resultHandle")]
+		IMPSHandle ResultHandle { get; }
+
+		[Export ("outputStateIsTemporary")]
+		bool OutputStateIsTemporary { get; set; }
+
+		[Export ("destinationImageAllocator", ArgumentSemantic.Retain)]
+		IMPSImageAllocator DestinationImageAllocator { get; set; }
+
+		[Export ("encodeToCommandBuffer:sourceImages:sourceStates:intermediateImages:destinationStates:")]
+		MPSImage Encode (IMTLCommandBuffer commandBuffer, MPSImage[] sourceImages, [NullAllowed] MPSState[] sourceStates, [NullAllowed] NSMutableArray<MPSImage> intermediateImages, [NullAllowed] NSMutableArray<MPSState> destinationStates);
+
+		[Export ("encodeToCommandBuffer:sourceImages:")]
+		MPSImage Encode (IMTLCommandBuffer commandBuffer, MPSImage[] sourceImages);
+
+		[Async, Export ("executeAsyncWithSourceImages:completionHandler:")]
+		MPSImage Execute (MPSImage[] sourceImages, Action<MPSImage, NSError> handler);
+	}
+
+	interface IMPSHandle { }
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface MPSHandle : NSCoding
+	{
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; }
+	}
+
+	interface IMPSCnnConvolutionDataSource { }
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject), Name = "MPSCNNConvolutionDataSource")]
+	interface MPSCnnConvolutionDataSource
+	{
+		[Abstract]
+		[Export ("dataType")]
+		MPSDataType DataType { get; }
+
+		[Abstract]
+		[Export ("descriptor")]
+		MPSCnnConvolutionDescriptor Descriptor { get; }
+
+		[Abstract]
+		[Export ("weights")]
+		IntPtr Weights { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("biasTerms")]
+		IntPtr /* float* */ BiasTerms { get; }
+
+		[Abstract]
+		[Export ("load")]
+		bool Load { get; }
+
+		[Abstract]
+		[Export ("purge")]
+		void Purge ();
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; }
+
+		[Export ("rangesForUInt8Kernel")]
+		IntPtr RangesForUInt8Kernel { get; }
+
+		[Export ("lookupTableForUInt8Kernel")]
+		IntPtr /* float* */ LookupTableForUInt8Kernel { get; }
+	}
+
+	interface IMPSNnPadding { }
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject), Name = "MPSNNPadding")]
+	interface MPSNnPadding : NSSecureCoding
```

Comment:
Two letters acronym, e.g. `NN`, stays capitalized.
It's only 3+ letters acronyms that needs to be pascal Cased.

Unless there's a `[Model]` then `[Protocol]` do not use `[BaseType]`